scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt

```diff
@@ -42,23 +43,54 @@ internal fun findDatasetsAndConnectors(
           ?.filter { it.varType == PARAMETERS_DATASET_ID }
           ?.forEach { parameter ->
             val parameterValue = scenario.parametersValues?.find { it.parameterId == parameter.id }
-            if (parameterValue != null && parameterValue.value != "") {
-              addDatasetAndConnector(
-                  datasetService,
-                  connectorService,
-                  organizationId,
-                  parameterValue.value,
-                  datasets,
-                  connectors,
-              )
-            }
+            addParameterValue(
+                parameterValue,
+                datasetService,
+                connectorService,
+                organizationId,
+                datasets,
+                connectors,
+            )
           }
     }
   }
   return DatasetsConnectors(
       datasets = datasets.values.toList(), connectors = connectors.values.toList())
 }
 
+private fun addParameterValue(
+    parameterValue: ScenarioRunTemplateParameterValue?,
+    datasetService: DatasetApiService,
+    connectorService: ConnectorApiService,
+    organizationId: String,
+    datasets: MutableMap<String, Dataset>,
+    connectors: MutableMap<String, Connector>,
+) {
+  if (parameterValue != null && parameterValue.value != "") {
+    val value = parameterValue.value
+    val datasetIdList = getDatasetIdListFromValue(value)
+    datasetIdList.forEach {
+      addDatasetAndConnector(
+          datasetService,
+          connectorService,
+          organizationId,
+          it,
+          datasets,
+          connectors,
+      )
+    }
+  }
+}
+
+internal fun getDatasetIdListFromValue(paramValue: String): List<String> {
+  if (paramValue.startsWith("[")) {
+    val datasetIds = paramValue.substring(1, paramValue.length - 1)
```

Comment:
This will throw an Index-out-of-bounds exception if `paramValue` is intentionally set to the one-character `[` String. Perhaps you could also make sure `paramValue` minimum length is 2.
This is actually what you are trying to achieve here, i.e. extract everything between `[` and the last character.

Reply:
Done with validation of last char = ']' + unit test